mvvm-test/src/main/java/com/thefuntasty/mvvm/test/BaseViewModelExtensionsMock.kt

```diff
@@ -0,0 +1,77 @@
+package com.thefuntasty.mvvm.test
+
+import androidx.lifecycle.LiveData
+import com.thefuntasty.mvvm.BaseViewModel
+import com.thefuntasty.mvvm.livedata.DefaultValueLiveData
+import com.thefuntasty.mvvm.livedata.DefaultValueMediatorLiveData
+import io.mockk.CapturingSlot
+import io.mockk.every
+import io.mockk.invoke
+
+/**
+ * Extension function that helps with mocking of [BaseViewModel.observeWithoutOwner]
+ *
+ * @return Captured lambda that is passed to original method. [invoke] should be called in order to simulate view state change.
+ *
+ * Usage:
+ *
+ * // GIVEN
+ * val capturedObserveLambda = viewModel.mockObserveWithoutOwner { viewState.number }
+ *
+ * // WHEN
+ * capturedObserveLambda.invoke(...)
+ *
+ */
+fun <VALUE> BaseViewModel<*>.mockObserveWithoutOwner(liveData: () -> LiveData<VALUE>): (VALUE) -> Unit {
+    var invokable: (CapturingSlot<(VALUE) -> Unit>)? = null
+    every { liveData().observeWithoutOwner(captureLambda()) } answers {
+        invokable = lambda()
+    }
+    return { value -> invokable?.invoke(value) ?: lambdaNotCapturedError() }
+}
+
+/**
+ * Extension function that helps with mocking of [BaseViewModel.observeWithoutOwner]
+ *
+ * @return Captured lambda that is passed to original method. [invoke] should be called in order to simulate view state change.
+ *
+ * Usage:
+ *
+ * // GIVEN
+ * val capturedObserveLambda = viewModel.mockObserveWithoutOwner { viewState.number }
+ *
+ * // WHEN
+ * capturedObserveLambda.invoke(...)
+ *
+ */
+fun <VALUE : Any> BaseViewModel<*>.mockObserveWithoutOwnerDefaultValue(liveData: () -> DefaultValueLiveData<VALUE>): (VALUE) -> Unit {
+    var invokable: (CapturingSlot<(VALUE) -> Unit>)? = null
+    every { liveData().observeWithoutOwner(captureLambda()) } answers {
+        invokable = lambda()
+    }
+    return { value -> invokable?.invoke(value) ?: lambdaNotCapturedError() }
+}
+
+/**
```

Comment:
<table data-meta="generated_by_danger-pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/d5e40af744bcbbf3aa0ef990344655ab4a8f443b/mvvm-test/src/main/java/com/thefuntasty/mvvm/test/BaseViewModelExtensionsMock.kt#L55"></span>Function names should be at most 35 characters long.</td>
    </tr>
  </tbody>
</table>